### DISCUSSION TITLE
[Chapter 02] private 필드나 메서드에 대한 Javadoc 처리

### DISCUSSION THREAD

**Discussion by corock** [cowaine/is-my-code-that-strange#16] [Chapter 01-03 (설계)] [answered]
2.1 ~ 2.3 장의 내용을 담은 [`Damage`](https://github.com/cowaine/is-my-code-that-strange/blob/feature/corock/corock/src/main/java/com/cowaine/corock/chapter02/game/Damage.java) 라는 클래스를 만들고 메서드에 대한 문서화 주석을 작성했습니다.
그리고 작성한 문서화 주석으로 `javadoc` 명령어를 이용하면 아래와 같이 정적 문서를 생성할 수 있습니다.

| Javadoc Summary                                                                                                   | Javadoc Details                                                                                                   |
|-------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
| ![image](https://github.com/cowaine/is-my-code-that-strange/assets/38161720/f63aeca9-ebbb-4dd8-af95-dc538939476a) | ![image](https://github.com/cowaine/is-my-code-that-strange/assets/38161720/2f0907e3-fcc1-423d-9faa-78964ca36167) |

생성한 정적 문서에는 공개된(`public`) 속성과 기능에 대해서만 클래스 명세가 되어 있어 `private` 메서드에 대한 문서화 주석이 보이지 않는데,
그럼 실무 등의 환경에서 `private` 접근 제어자에 문서화 주석을 달 필요는 없을까요? 아래는 `private` 필드에 문서화 주석을 작성한 예시입니다.

```java
public class ContractAmount {

    /**
     * 세금 포함 금액
     */
    private final int amountIncludingTax;

    /**
     * 소비세율
     */
    private final BigDecimal salesTaxRate;

    // (...)
}
```

**Comment by Irisation23** [comment]
### 이슈 사항

- Damage 클래스 링크 404 에러발생

### 원인

- corock 브랜치 삭제로 인한 영향

<img width="823" alt="image" src="https://github.com/cowaine/is-my-code-that-strange/assets/53285909/635bbbb4-a974-45b5-8023-d136e74e7d7d">

### 해결방안

하위 링크 참조

- https://github.com/cowaine/is-my-code-that-strange/blob/develop/corock/src/main/java/com/cowaine/corock/chapter02/game/Damage.java

**Comment by firewoody237** [comment]
캡슐화로 클래스의 메시지만 보내면 되지 않을까요?
문서화는 해당 클래스의 사용자를 위한 것인데 외부 사용자가 private까지는 알 필요가 없지 않을까 싶습니다.

**Comment by coalong** [comment, marked as answer]
찾아보니, JavaDoc 을 생성할 경로 및 포함시킬 접근지정자(private, package, protected, public …)을 설정할 수 있는 것 같네요.

저는 되도록 모든 JavaDoc 을 작성하고 필요에 따라 사용하는 JavaDoc 을 설정하면 좋을 것 같습니다.
